src/main/scala/cognite/spark/v1/FlexibleDataModelBaseRelation.scala

```diff
@@ -56,36 +56,65 @@ abstract class FlexibleDataModelBaseRelation(config: RelationConfig, sqlContext:
     })
   }
 
+  // scalastyle:off method.length
   // scalastyle:off cyclomatic.complexity
-  private def extractInstancePropertyValue: InstancePropertyValue => Any = {
-    case InstancePropertyValue.String(value) => value
-    case InstancePropertyValue.Int32(value) => value
-    case InstancePropertyValue.Int64(value) => value
-    case InstancePropertyValue.Float32(value) => value
-    case InstancePropertyValue.Float64(value) => value
-    case InstancePropertyValue.Boolean(value) => value
-    case InstancePropertyValue.Date(value) => java.sql.Date.valueOf(value)
-    case InstancePropertyValue.Timestamp(value) => java.sql.Timestamp.from(value.toInstant)
-    case InstancePropertyValue.Object(value) => value.noSpaces
-    case InstancePropertyValue.ViewDirectNodeRelation(value) =>
-      value.map(r => Array(r.space, r.externalId)).orNull
-    case InstancePropertyValue.StringList(value) => value
-    case InstancePropertyValue.BooleanList(value) => value
-    case InstancePropertyValue.Int32List(value) => value
-    case InstancePropertyValue.Int64List(value) => value
-    case InstancePropertyValue.Float32List(value) => value
-    case InstancePropertyValue.Float64List(value) => value
-    case InstancePropertyValue.DateList(value) => value.map(v => java.sql.Date.valueOf(v))
-    case InstancePropertyValue.TimestampList(value) =>
-      value.map(v => java.sql.Timestamp.from(v.toInstant))
-    case InstancePropertyValue.ObjectList(value) => value.map(_.noSpaces)
-    case InstancePropertyValue.TimeSeriesReference(value) => value
-    case InstancePropertyValue.FileReference(value) => value
-    case InstancePropertyValue.SequenceReference(value) => value
-    case InstancePropertyValue.TimeSeriesReferenceList(value) => value
-    case InstancePropertyValue.FileReferenceList(value) => value
-    case InstancePropertyValue.SequenceReferenceList(value) => value
+  private def extractInstancePropertyValue(key: String, value: InstancePropertyValue): Any = {
```

Comment:
looks like this one should be the inverse of `deriveViewPropertySchemaWithUsageSpecificAttributes`, not sure if there's an easier way to construct them both, will try to cross-check at least

Reply:
yes deriveViewPropertySchemaWithUsageSpecificAttributes derives the sparc schema from the view. Here we use it to set the "correct" type from json..